alf/environments/suite_metadrive.py

```diff
@@ -126,6 +158,10 @@ def _acquire_next_frame(self, action):
         """
         observation, reward, done, info = self._env.step(action)
 
+        # Apply extra rewards
+        extra_reward, extra_env_info = self._compute_extra_reward()
+        reward += extra_reward
+
```

Comment:
To reviewers: the change here can be used as an entry point to read the whole diff. Here extra rewards are computed every time when the underlying environment `step()`.